auth/token.go

```diff
@@ -30,6 +37,70 @@ const (
 	UsernameClaimKey = "username"
 )
 
+func init() {
+	// ensures we don't generate predictable token signing keys
+	rand.Seed(time.Now().UnixNano())
+}
+
+var tokenCharacters = []rune("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890")
+
+func generateTokenSigningKey() string {
```

Comment:
What do you think of following approach 3 which seems to be fast http://stackoverflow.com/questions/22892120/how-to-generate-a-random-string-of-a-fixed-length-in-golang

And, this func can be named `generateRandString`. Yay??